packages/backend/src/services/UserAttributesService/UserAttributesService.ts

```diff
@@ -0,0 +1,82 @@
+import { subject } from '@casl/ability';
+import {
+    CreateOrgAttribute,
+    ForbiddenError,
+    OrgAttribute,
+    SessionUser,
+} from '@lightdash/common';
+import { UserAttributesModel } from '../../models/UserAttributesModel';
+
+type Dependencies = {
+    userAttributesModel: UserAttributesModel;
+};
+
+export class UserAttributesService {
+    private readonly userAttributesModel: UserAttributesModel;
+
+    constructor(dependencies: Dependencies) {
+        this.userAttributesModel = dependencies.userAttributesModel;
+    }
+
+    async getAll(user: SessionUser): Promise<OrgAttribute[]> {
+        return this.userAttributesModel.find({
+            organizationUuid: user.organizationUuid!,
+        });
+    }
+
+    async create(
+        user: SessionUser,
+        orgAttribute: CreateOrgAttribute,
+    ): Promise<OrgAttribute> {
+        if (user.ability.cannot('manage', 'Organization')) {
+            throw new ForbiddenError();
+        }
+        return this.userAttributesModel.create(
+            user.organizationUuid!,
+            orgAttribute,
+        );
+    }
+
+    async update(
+        user: SessionUser,
+        orgAttributeUuid: string,
+        orgAttribute: CreateOrgAttribute,
+    ): Promise<OrgAttribute> {
+        const savedAttribute = await this.userAttributesModel.get(
+            orgAttributeUuid,
+        );
+
+        if (
+            user.ability.cannot(
+                'manage',
```

Comment:
Optional, but after recent refactoring of permissions, I'm wondering if we add new subjects for things we might later want to permission differently.

For example this could be
```typescript
cannot('update', subject('UserAttribute', {organizationUuid, attribute.organizationUuid}))
```

That way we have explicit permissions throughout the app, and we only have to change them by updating the ability definitions files

Reply:
I think this is better to implement when we need it, otherwise, permissions can grow very quickly and become  very hard to maintain.